src/main/java/jp/co/future/uroborosql/SqlEntityQueryImpl.java

```diff
@@ -75,11 +97,1002 @@
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#notLike(java.lang.String, java.lang.CharSequence)
+	 */
+	@Override
+	public SqlEntityQuery<E> notLike(final String col, final CharSequence searchValue) {
+		context().param(col, new NotLike(col, searchValue));
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#notStartsWith(java.lang.String, java.lang.CharSequence)
+	 */
+	@Override
+	public SqlEntityQuery<E> notStartsWith(final String col, final CharSequence searchValue) {
+		String escaped = agent().getSqlConfig().getDialect().escapeLikePattern(searchValue);
+		context().param(col, new NotLike(col, escaped, true));
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#notEndsWith(java.lang.String, java.lang.CharSequence)
+	 */
+	@Override
+	public SqlEntityQuery<E> notEndsWith(final String col, final CharSequence searchValue) {
+		String escaped = agent().getSqlConfig().getDialect().escapeLikePattern(searchValue);
+		context().param(col, new NotLike(col, true, escaped));
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#notContains(java.lang.String, java.lang.CharSequence)
+	 */
+	@Override
+	public SqlEntityQuery<E> notContains(final String col, final CharSequence searchValue) {
+		String escaped = agent().getSqlConfig().getDialect().escapeLikePattern(searchValue);
+		context().param(col, new NotLike(col, true, escaped, true));
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#between(java.lang.String, java.lang.Object, java.lang.Object)
+	 */
+	@Override
+	public SqlEntityQuery<E> between(final String col, final Object fromValue, final Object toValue) {
+		context().param(col, new Between(col, fromValue, toValue));
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#isNull(java.lang.String)
+	 */
+	@Override
+	public SqlEntityQuery<E> isNull(final String col) {
+		context().param(col, new IsNull(col));
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#isNotNull(java.lang.String)
+	 */
+	@Override
+	public SqlEntityQuery<E> isNotNull(final String col) {
+		context().param(col, new IsNotNull(col));
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#where(java.lang.CharSequence)
+	 */
+	@Override
+	public SqlEntityQuery<E> where(final CharSequence rawString) {
+		this.rawString = rawString;
+		return this;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 *
+	 * @see jp.co.future.uroborosql.fluent.SqlEntityQuery#orderByAsc(java.lang.String[])
+	 */
+	@Override
+	public SqlEntityQuery<E> orderByAsc(final String... cols) {
+		this.orderByColumns = Arrays.asList(cols);
```

Comment:
I'd like you to call `orderByAsc` and `orderByDesc`, multiple times, too.
Can not correspond to `nulls first `and `nulls last`?

Reply:
Change the API as follows. And asc () / desc () to be called multiple times.

```
/**
 * multi columns. nulls last(default)
 */
asc(String... cols)
```
```
/**
 * single column. pass nulls option(Nulls.FIRST / Nulls.LAST). 
 */
asc(String col, Nulls nulls)
```